File: app/src/main/java/com/example/verviapp/data/entity/ServiceApplicationEntity.kt
package com.example.verviapp.data.entity

import androidx.room.Entity
import androidx.room.ForeignKey
import androidx.room.Index
import androidx.room.PrimaryKey

@Entity(
    tableName = "service_applications",
    foreignKeys = [
        ForeignKey(
            entity = RequestEntity::class,
            parentColumns = ["id"],
            childColumns = ["requestId"],
            onDelete = ForeignKey.CASCADE
        ),
        ForeignKey(
            entity = UserEntity::class,
            parentColumns = ["id"],
            childColumns = ["providerUserId"],
            onDelete = ForeignKey.CASCADE
        )
    ],
    indices = [
        Index(value = ["requestId"]),
        Index(value = ["providerUserId"]),
        Index(value = ["status"])
    ]
)
data class ServiceApplicationEntity(
    @PrimaryKey(autoGenerate = true)
    val id: Int = 0,
    val requestId: Int,
    val providerUserId: Int,
    val presentationMessage: String,
    val proposedPriceCop: Long? = null,
    val evidenceUri: String? = null,
    val immediateAvailability: Boolean = false,
    val status: String = "PENDING",
    val createdAt: Long = System.currentTimeMillis(),
    val updatedAt: Long = System.currentTimeMillis()
)

File: vervi-api/src/service-applications/service-application.entity.ts
import { Entity, PrimaryGeneratedColumn, Column, CreateDateColumn, UpdateDateColumn, ManyToOne, JoinColumn, OneToMany } from 'typeorm';
import { Request } from '../requests/request.entity';
import { User } from '../users/user.entity';
import { Notification } from '../notifications/notification.entity';

@Entity('service_applications')
export class ServiceApplication {
  @PrimaryGeneratedColumn()
  id: number;

  @Column()
  requestId: number;

  @ManyToOne(() => Request, request => request.applications, { onDelete: 'CASCADE' })
  @JoinColumn({ name: 'requestId' })
  request: Request;

  @Column()
  providerUserId: number;

  @ManyToOne(() => User, user => user.providerServices, { onDelete: 'CASCADE' })
  @JoinColumn({ name: 'providerUserId' })
  provider: User;

  @Column({ type: 'text' })
  presentationMessage: string;

  @Column({ nullable: true })
  proposedPriceCop: number;

  @Column({ nullable: true })
  evidenceUri: string;

  @Column({ default: false })
  immediateAvailability: boolean;

  @Column({
    type: 'enum',
    enum: ['PENDING', 'ACCEPTED', 'REJECTED'],
    default: 'PENDING'
  })
  status: string;

  @OneToMany(() => Notification, notification => notification.application)
  notifications: Notification[];

  @CreateDateColumn()
  createdAt: Date;

  @UpdateDateColumn()
  updatedAt: Date;
}
